### PR TITLE
Fix regression in analyzer partial analysis state tracking

### LINKED ISSUE #63466: Quick fix is suggesting that I make properties that access instance data static
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Quick-fix-is-suggesting-that-I-make-prop/10122238)._

---
[severity:It bothers me. A fix would be nice] [regression]
1. Create a new Console App
2. Choose .NET 6.0 (LTS)
3. Paste the following code:

```public class Example
{
    public string Name { get; set; }

    public string AccessInstance => this.Name + "Foo";
}
```

**Result**

AccessInstance has a Quick Fix teaser on it that suggests making the property static which would lead to build errors.

**Expected**

No quick fix suggestion for AccessInstance around making it static.

---
#### Feedback Bot on 8/16/2022, 07:19 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #63466

My recent PR #63206 fixed partial analysis state tracking where we were making duplicate operation/syntax node callbacks into analyzers across multiple calls to CompilationWithAnalyzers.GetAnalysisResultAsync. However, that introduced a regression #63466 for analyzers that register operation actions both inside and outside an OperationBlockStartAnalysisContext. We execute the operation actions registered in OperationBlockStartAnalysisContext separately from those registered outside of it, but we were incorrectly sharing the same analyzer state for both. This led to skipping one set of analyzer actions. This PR fixes this regression by using separate states for these.

Note that we were already using separate states for syntax node actions registered inside and outside of CodeBlockStartAnalysisContext, so this bug does not affect syntax node based analyzers with similar registration pattern.

Verified that the added unit test fails prior to the fix when Operation actions are used in the test analyzer. The added unit test passes prior to the fix when SyntaxNode actions are used. Also verified that CA1822 regression scenarios in #63466 are also fixed with this change.

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs [hunk 1]
```diff
@@ -1449,7 +1449,7 @@ public void MarkSymbolEndAnalysisComplete(ISymbol symbol, DiagnosticAnalyzer ana
             {
                 if (TryStartAnalyzingDeclaration(declaredSymbol, declarationIndex, analyzer, analysisScope, analysisState, out analyzerState))
                 {
-                    ExecuteOperationActionsCore(operationsToAnalyze, operationActionsByKind, analyzer, declaredSymbol, model, filterSpan, analyzerState?.OperationBlockAnalysisState.ExecutableNodesAnalysisState, isGeneratedCode);
+                    ExecuteOperationActionsCore(operationsToAnalyze, operationActionsByKind, analyzer, declaredSymbol, model, filterSpan, analyzerState?.OperationAnalysisState, isGeneratedCode);
```
Comment (mavasani) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs:1452]: This is the core fix - we no longer share the same state for operation actions registered inside OperationBlockStartAnalysisContext and those registered outside of it

**Review by 333fred — APPROVED** [review]

**Review by genlu — APPROVED** [review]

**Comment by mavasani** [comment]
Merging this to help get this inserted asap into VS. If there is any additional feedback, I’ll address it with a follow up PR.

**Comment by mavasani** [comment]
@dibarbet Is it possible to fast track this change into the next Roslyn insertion?

**Comment by dibarbet** [comment]
> @dibarbet Is it possible to fast track this change into the next Roslyn insertion?

the next merge will contain this PR (actively working on some cloudbuild fixes).

**Pull request merged**